pkg/controller/job/job_controller_test.go

```diff
@@ -1737,10 +1741,6 @@ func TestPastDeadlineJobFinished(t *testing.T) {
 			if err := sharedInformerFactory.Batch().V1().Jobs().Informer().GetIndexer().Add(job); err != nil {
 				t.Fatalf("Failed to insert job in index: %v", err)
 			}
-			// This is needed because the fake clientset doesn't report created pods
-			// to the informer, leading to unsatisfied expectations in syncJob that prevent the controller from setting the final condition.
-			podIndexer := sharedInformerFactory.Core().V1().Pods().Informer().GetIndexer()
-			setPodsStatuses(podIndexer, job, 0, 1, 0, 0, 0)
 
 			var j *batch.Job
 			err = wait.PollImmediate(200*time.Microsecond, 3*time.Second, func() (done bool, err error) {
```

Comment:
You can get rid of the checks in lines 1756 to 1761 and put it inside the Poll, if `clock.Now()` is earlier than 1s.

Reply:
Done